neuralmonkey/dataset.py

```diff
@@ -395,3 +407,22 @@ def shuffle(self):
     def series_ids(self) -> Iterable[str]:
         return (list(self.series_paths_and_readers.keys()) +
                 list(self.preprocess_series.keys()))
+
+    def add_series(self, name: str, series: Iterable[Any]) -> None:
+        raise NotImplementedError(
+            "Lazy dataset does not support adding series.")
+
+
+def _preprocessed_datasets(dataset: Dataset, kwargs: Dict[str, Any]) -> None:
```

Comment:
Bylo by to lepší posunout, ale to by se vevnitř nesmělo používat `Dataset` a `LazyDataset` vevnitř.